runtimes/core/src/api/cors/mod.rs

```diff
@@ -62,21 +75,38 @@ pub fn config(cfg: &pb::gateway::Cors, meta: MetaHeaders) -> anyhow::Result<Cors
             }
             _ => OriginSet::Some(vec![]),
         };
-        let without_creds = OriginSet::new(
-            cfg.allowed_origins_without_credentials
-                .clone()
-                .unwrap_or_default()
-                .allowed_origins,
-        );
+        let without_creds = {
+            if let Some(CorsAllowedOrigins { allowed_origins }) =
+                &cfg.allowed_origins_without_credentials
+            {
+                OriginSet::new(allowed_origins.to_vec())
+            } else {
+                OriginSet::All
+            }
+        };
+
+        let request_has_creds = |req: &axum::http::request::Parts| -> bool {
+            if req.headers.contains_key(AUTHORIZATION) || req.headers.contains_key(COOKIE) {
+                return true;
+            }
+
+            if req.method == http::method::Method::OPTIONS {
+                return req
+                    .headers
+                    .get(ACCESS_CONTROL_REQUEST_HEADERS)
+                    .and_then(|val| val.to_str().ok())
+                    .map(|val| val.to_lowercase().contains("authorization"))
```

Comment:
Should we check for `cookie` here, too?

Reply:
I guess so since we check it on the "real" request, added it